### PR TITLE
workflow to update discussions data files added

### LINKED ISSUE #6774: [CI] Docs: Create workflow to refresh list of discussion forum posts
#### Current Behavior
Meshery Docs displays the most recent 10 discussion forum posts by tag name. Currently "mesheryctl" and "meshery" are the two supported tags. See this related PR https://github.com/meshery/meshery/pull/6773 and this discussion https://discuss.layer5.io/t/proposal-and-call-for-participation-building-the-community-leaderboard/1712/2

#### Desired Behavior
A scheduled (cron) workflow should refresh the multiple lists of related discussion data files (json) on a daily basis.

#### Implementation
1. Checkout repo
1. Pull fresh list of related posts from:
    1. http://discuss.layer5.io/tag/mesheryctl.json
    1. http://discuss.layer5.io/tag/meshery.json
1. Deposit data files here - https://github.com/meshery/meshery/tree/master/docs/_data/discuss
1. Autocommit fresh data files.

#### Acceptance Tests
1. Fresh list of related posts display in the Recent Discussions with "<tag>" Tag (e.g. https://docs.meshery.io/guides/mesheryctl)

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by deepakdinesh1123**
Signed-off-by: deepakdinesh1123 <d.deepakdinesh13@gmail.com>

**Notes for Reviewers**

This PR fixes #6774
@leecalcote please review this PR and tell me any changes that I need to make to my approach

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Chadha93** [comment]
@deepakdinesh1123  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by leecalcote** [comment]
> @deepakdinesh1123 Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

That would be great.

File: .github/workflows/discussions-data-files-update.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Fetch data for mesheryctl tag
+        run: curl https://discuss.layer5.io/tag/mesheryctl.json > docs/_data/discuss/mesheryctl.json
+      - name: Fetch data for meshery tag
+        run: curl https://discuss.layer5.io/tag/meshery.json > docs/_data/discuss/meshery.json
```
Comment (leecalcote) [inline .github/workflows/discussions-data-files-update.yml:16]: I wonder if `-o` as the `curl` native flag is perhaps ideal vs. `>`

File: .github/workflows/discussions-data-files-update.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Commit results
+        run: |
+          git config user.name "Discussions Bot"
+          git config user.email "discussions_bot@github.com"
```
Comment (leecalcote) [inline .github/workflows/discussions-data-files-update.yml:20]:
```suggestion
          git config user.email "discussions@meshery.io"
```

**Review by leecalcote — CHANGES_REQUESTED** [review]
I have a hunch that we'll ultimately need to leverage something like this autocommit action ([example use](https://github.com/meshery/meshery/blob/8ff11b8cb5659bd3842424d2df6cbbf1e6232af6/.github/workflows/integrations-updater.yml#L36-L44)) in order to ensure that the `git` operation is empowered with the appropriate level of permissions to `git push` and override the `master` branch protections.

**Comment by leecalcote** [comment]
Some good stuff here, @deepakdinesh1123. Thank you.

**Comment by leecalcote** [comment]
@deepakdinesh1123 does this feedback make sense?

**Comment by deepakdinesh1123** [comment]
@leecalcote I am sorry as I could not update this as I had some uni work and exams I will incorporate the suggested changes by Monday

**Comment by leecalcote** [comment]
> @leecalcote I am sorry as I could not update this as I had some uni work and exams I will incorporate the suggested changes by Monday

Understood. Did you end up opening a new PR?

**Comment by deepakdinesh1123** [comment]
@leecalcote I have opened a new PR #6832 as I did a mistake when syncing new changes

**Pull request closed without merging**